### ISSUE TITLE
System.exit()

### ISSUE THREAD

**Issue by fprochazka** [uber/NullAway#314]
I've found a case, that is not handled properly (I think). I expect the JVM to shutdown when I call `System.exit(1);` and therefore I'd expect this statement to behave as a control-flow guard, similar to putting there a `return;`. But it is ignored.

```java
@Nullable CommandsGroup commandBean = getCommandBean(commandNameParts[0], context);
if (commandBean == null) {
    System.err.println("Command " + commandName + " not found.");
    System.exit(1);
}

commandBean.runCommand(commandNameParts[1], context.getBean(ApplicationArguments.class));
```

produces

```
[ERROR] ConsoleApplication.java:[54,24] [NullAway] dereferenced expression commandBean is @Nullable
    (see http://t.uber.com/nullaway )
```

The fix in my code is trivial

```java
@Nullable CommandsGroup commandBean = getCommandBean(commandNameParts[0], context);
if (commandBean == null) {
    System.err.println("Command " + commandName + " not found.");
    System.exit(1);

} else {
    commandBean.runCommand(commandNameParts[1], context.getBean(ApplicationArguments.class));
}
```

But IMHO, think this is something the NullAway should handle.

I wasn't looking very hard, but I don't believe there is a way to escape the `System.exit(1);` (apart from the shutdown handler hook, that is executed in different thread and there doesn't affect the method and thread calling the exit).

**Comment by msridhar** [comment]
You're right that NullAway doesn't currently model `System.exit()` correctly.  I think using it outside of a main driver class is regarded as an anti-pattern since, as you point out, there is no easy way for calling code to stop or catch it.  We don't have a straightforward way to model this right now, but if we get time we can look into it.  I wouldn't consider it particularly high priority given the rarity of use.  Hopefully your workarounds are sufficient for now.
